biz.aQute.resolve/src/biz/aQute/resolve/ResolverValidator.java

```diff
@@ -65,21 +70,36 @@ public void setSystem(Resource resource) throws Exception {
 	}
 
 	public List<Resolution> validate() throws Exception {
+		FixedIndexedRepo repository = getRepository();
+		Set<Resource> resources = getAllResources(repository);
+		return validateResources(repository, resources);
+	}
+
+	public List<Resolution> validate(Collection<Resource> toBeChecked) throws Exception {
+		Set<Resource> allResources = new LinkedHashSet<Resource>();
+		for (URI uri : repositories) {
+			allResources.addAll(XMLResourceParser.getResources(uri));
+		}
+		allResources.addAll(toBeChecked);
+		ResourcesRepository repository = new ResourcesRepository(allResources);
+		return validateResources(repository, toBeChecked);
+	}
 
+	FixedIndexedRepo getRepository() throws MalformedURLException, URISyntaxException {
 		FixedIndexedRepo repository = new FixedIndexedRepo();
 		repository.setLocations(Strings.join(repositories));
-
-		Set<Resource> resources = getAllResources(repository);
-		setProperty("-runfw", "dummy");
-
-		return validateResources(repository, resources);
+		return repository;
 	}
 
-	public List<Resolution> validateResources(Repository repository, Set<Resource> resources) throws Exception {
+	public List<Resolution> validateResources(Repository repository, Collection<Resource> resources) throws Exception {
+		setProperty("-runfw", "dummy");
 		List<Resolution> result = new ArrayList<>();
-		for (Resource resource : resources) {
+		List<Resource> resourceList = new ArrayList<>(resources);
+		while (!resourceList.isEmpty()) {
+			Resource resource = resourceList.remove(0);
 			Resolution resolution = resolve(repository, resource);
 			result.add(resolution);
```

Comment:
While this change is good for performance, I think it materially change the semantics of the method. Before if you passed it N Resources, you got back a list of N Resolutions (one per Resource). This change violates that. However, I think you can fix that somewhat by creating Resolutions for each element in resolution.resolved.